cmd/kube-scheduler/app/server.go

```diff
@@ -61,6 +62,18 @@ import (
 // Option configures a framework.Registry.
 type Option func(runtime.Registry) error
 
+// TODO: delete this check after insecure flags removed in v1.24
+func checkNonZeroInsecurePort(fs *pflag.FlagSet) error {
+	val, err := fs.GetInt("port")
+	if err != nil {
+		return err
+	}
+	if val != 0 {
+		return fmt.Errorf("invalid port value %d: only zero is allowed", val)
```

Comment:
just to make sure: what is the point of validating that the flag is zero?

Is this the effect we want: people who were using the flag need to stop using. But people that were using the flag to disable the feature can keep it?

Reply:
This is what was done for the other components. The rationale is indeed to force them to migrate.

Reply:
gotcha

/approve